### DISCUSSION TITLE
Swelljs is rendering duplicate paypal buttons

### DISCUSSION THREAD

**Discussion by Buupu** [swellstores/community#113] [Help] [answered]
I'm using swelljs to render a paypal payment button but I'm getting multiple buttons rendering despite only calling createElements once - 

![image](https://user-images.githubusercontent.com/26443341/214845265-2eabe0e3-aeb2-4479-9008-15eeb378018e.png)

Here is the code I'm using - 

```
useEffect(() => {
    SwellClient.payment.createElements({
      card: {
        elementId: "#card-element-id",
        options: {
          style: {
            base: {
              fontWeight: 500,
              fontSize: "16px",
            },
          },
        },
      },
      paypal: {
        elementId: "#paypal-button",
        style: {
          layout: "horizontal",
          label: "buynow",
          tagline: false,
        },
      },
    });
  }, []);
```

With `<div id="paypal-button" />` as my HTML element.

Any help with this would be appreciated!

**Comment by ericingram** [comment]
Could `useEffect` be getting called twice? I don't think it clears an existing element so that would make sense.

**Reply by Buupu** [reply, marked as answer]
Only triggering once according to my logs. Weirdly enough, if I attach the function to an event instead then it doesn't duplicate the elements. Probably some hot reload issue I'd imagine. 

Happy to close this.

**Comment by sags95** [comment]
@Buupu  is `StrictMode` enabled by any chance?

**Reply by Buupu** [reply]
Yes, it is. Could that be the cause?

**Comment by PovilJ** [comment]
Got same problem

**Comment by rinorzek** [comment]
Same problem as well.
